src/main/java/org/matsim/prepare/PrepareTransitSchedule.java

```diff
@@ -101,70 +90,43 @@ public void run(Scenario scenario, String shapeFile) {
 
     private String findFareZone(TransitStopFacility transitStopFacility, Collection<SimpleFeature> features) {
 
-        String fareZone = "";
-        Boolean stopInShapes = false;
+        // Facilities which are not whithin the fare zone shapes are located outside of vvs area and thus marked accordingly
+        String fareZone = "out";
 
         Coord homeCoord = transitStopFacility.getCoord();
         Point point = MGC.coord2Point(homeCoord);
 
-        // Automatic crs detection/ transformation might be needed to avoid errors.
-
         for (SimpleFeature feature : features ) {
 
             Geometry geometry = (Geometry) feature.getDefaultGeometry();
 
             if (geometry.contains(point)) {
                 fareZone = feature.getAttribute("FareZone").toString();
-                stopInShapes = true;
             }
 
         }
 
-
-        if (! stopInShapes) {
-
-            // There are several home locations outside the shape File boundaries
-            // as for performance reasons the shape File was reduced to Stuttgart Metropolitan Area
-
-            // How to deal with the people that do not live in Stuttgart Metropolitan Area?
-            // Will we need more detailed information on their home locations as well?
-
-            //log.info("No fareZone found for transit stop facility with id: " + transitStopFacility.getId());
-        }
-
         return fareZone;
 
     }
 
 
     private List<Id<TransitStopFacility>> tagBikeAndRide(TransitSchedule schedule) {
 
-        // The idea is that all stops with departures of tram and train within vvs area have BikeAndRideFacilities
-        // and allow Bike and Ride
+        // This method writes all stops in a list that have departures of mode "tram" or "train"
+        // It is assumed that at tram or train stations Bike-and-Ride is possible
 
         List<Id<TransitStopFacility>> bikeAndRide = new ArrayList<>();
-        List<String> modes = Arrays.asList(new String[]{"tram", "train"});
-
-        schedule.getTransitLines().values().stream().forEach(transitLine -> {
-
-            transitLine.getRoutes().values().stream().forEach(transitRoute -> {
+        List<String> modes = Arrays.asList("tram", "train");
 
-                if (modes.contains(transitRoute.getTransportMode())){
```

Comment:
Either use a nested for loop i.e. 

```
for (var line: schedule.getTransitLines().values()) {
    for(var route : line.getRoutes().vaules()) {
        // do stuff
    }
}
```

or use flat map if you want to use streams: 

```
var bikeAndRide = schedule.getTransitLInes().values().stream()
    .flatMap(line -> line.getRoutes.values().stream())
    .filter(route -> modes.contains(route.getTransportMode())
    .flatMap(route -> route.getStops().stream())
    .map(stop -> stop.getId())
    .collect(Collectors.toSet());
```
I personally would prefere the second one

Reply:
Chose the second one.